chain/client/src/test_utils.rs

```diff
@@ -1486,14 +1486,20 @@ pub fn run_catchup(
 ) -> Result<Vec<AcceptedBlock>, Error> {
     let mut result = vec![];
     let f = |_| {};
-    let messages = Arc::new(RwLock::new(vec![]));
-    let inside_messages = messages.clone();
+    let block_messages = Arc::new(RwLock::new(vec![]));
+    let block_inside_messages = block_messages.clone();
     let block_catch_up = move |msg: BlockCatchUpRequest| {
-        inside_messages.write().unwrap().push(msg);
+        block_inside_messages.write().unwrap().push(msg);
     };
+    let state_split_messages = Arc::new(RwLock::new(vec![]));
+    let state_split_inside_messages = state_split_messages.clone();
+    let state_split = move |msg: StateSplitRequest| {
+        state_split_inside_messages.write().unwrap().push(msg);
+    };
+    let rt = client.runtime_adapter.clone();
     while !client.chain.store().iterate_state_sync_infos().is_empty() {
-        let call = client.run_catchup(highest_height_peers, &f, &block_catch_up)?;
-        for msg in messages.write().unwrap().drain(..) {
+        let call = client.run_catchup(highest_height_peers, &f, &block_catch_up, &state_split)?;
```

Comment:
I didn't appreciate why you put the logic for sending requests to a closure until now. This makes testing way easier. Great design!